### DISCUSSION TITLE
Are the tools to create new languages still usable?

### DISCUSSION THREAD

**Discussion by Wysardry** [davidgiven/ack#295] [Q&A] [answered]
If I wanted to add a new programming language to ACK, would I be able to do that using the code that is currently available?

**Comment by davidgiven** [comment, marked as answer]
Yes; they're a bit archaic, but perfectly usable. You'll want a compiler that's oriented around stack-based code generation, i.e., nearly all of them. A while back I integrated this B compiler: https://github.com/aap/abc It was mostly straightforward, although making the system handle 16-bit bytes was a bit painful.
